### PR TITLE
revert [SPARK-10030] Use tags to control which tests to run depending on changes being tested

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

This PR reverts https://github.com/apache/spark/pull/8775

The problem is that, when we change the code in Spark core, we may break yarn test, so we should run yarn test even we didn't change code in the yarn module. We broke the build because of this issue in https://github.com/apache/spark/pull/20223

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @vanzin @sameeragarwal

**Comment by JoshRosen** [comment]
Are you sure that we want to blanket revert this entire patch? Is there a more surgical short-term fix we can make in `dev/sparktestsupport/modules.py` to just always unconditionally enable the tag for now?

Also, is this the first time recently that we've failed the YARN integration tests? How much time do they add?

The trade off here seems to be between slightly slower after-the-fact detection of a test failure / build break due to YARN vs. faster tests for the majority of PRs that don't touch YARN code. I think we've had one or two such breaks in the 2+ years that we've been using these test tags, so I'd also be fine with postponing this change if you agree that it's unlikely that we're going to have many such failures here.

If the motivation is that it's hard to test the fix for such build breaks (because the failing test wouldn't be exercised in the PR builder) then I think we might already have a solution via special tags placed into the PR title (I think `test-yarn` or something similar; see `run-tests-jenkins.py`).

**Comment by cloud-fan** [comment]
We want a way to test yarn for the new fix: https://github.com/apache/spark/pull/20297

I checked `run-tests-jenkins.py` and seems we only support `test-maven`, `test-hadoop2.6` and `test-hadoop2.6`. It will be good if we have a `test-yarn` and then we don't need this patch.

File: common/tags/src/test/java/org/apache/spark/tags/DockerTest.java [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-
-package org.apache.spark.tags;
-
-import java.lang.annotation.*;
-import org.scalatest.TagAnnotation;
-
-@TagAnnotation
-@Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.METHOD, ElementType.TYPE})
-public @interface DockerTest { }
```
Comment (cloud-fan) [inline common/tags/src/test/java/org/apache/spark/tags/DockerTest.java]: We don't use this docker tag in `modules.py`, does it mean we never run docker tests in PR builder?

Comment (cloud-fan) [reply common/tags/src/test/java/org/apache/spark/tags/DockerTest.java]: if it is, we should preserve this behavior.

**Comment by vanzin** [comment]
@cloud-fan 
> We want a way to test yarn for the new fix: 20297

In that case we're ok because that fix touches YARN code, so YARN tests are running.
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86289/testReport/org.apache.spark.deploy.yarn/

**Comment by vanzin** [comment]
To answer one of Josh's question, from the link above, it seems the YARN integration tests currently add ~5m to the PRB time. That doesn't sound too horrible, but would potentially make current timeout issues worse since we seem to be brushing against the existing limit already.

**Pull request closed without merging**